src/tcgwars/logic/impl/gen4/Stormfront.groovy

```diff
@@ -2926,38 +2926,53 @@ public enum Stormfront implements LogicCardInfo {
                   def pkmnCard = self.topPokemonCard
-                  def pcs=self
-                  delayed(inline: true){
-                    after KNOCKOUT, pcs, {
-                      def stadiumCard
-                      stadiumCard = stadium(new CustomCardInfo(DUSKNOIR_LV_X_96).setCardTypes(TRAINER, STADIUM)) {
-                        def eff
-                        bc "here"
-                        onPlay {
-                          bc "here 2"
-                          eff = delayedA {
-                            bc "here 3"
-                            before null, {
-                              bc "here 4"
-                            }
+                  def stadiumCard
+                  stadiumCard = stadium(new CustomCardInfo(DUSKNOIR_LV_X_96).setCardTypes(TRAINER, STADIUM)) {
+                    def eff
+                    onPlay {
+                      eff = delayed inline:true, {
+                        before BEGIN_TURN, {
+                          thisCard.player.opposite.pbg.all.each {
+                            directDamage 10, it, TRAINER_CARD
                           }
                         }
-                        onRemoveFromPlay {
-                          bc "here remove"
-                          eff.unregister()
-                          bg.em().run(new ChangeImplementation(pkmnCard, stadiumCard))
-                          moveCard(pkmnCard, thisCard.player.pbg.hand)
+                      }
+                      delayed inline:true, {
+                        after REMOVE_FROM_PLAY, {
+                          if (LUtils.isRemoveFromPlayAndContainsCard(e, stadiumCard)) {
+                            stadiumCard.onDiscard bg
+                            bg.setStadiumInfoStruct null
+                            unregister()
+                          }
                         }
                       }
-                      stadiumCard.player = thisCard.player
-                      bg.em().run(new ChangeImplementation(stadiumCard, pkmnCard))
-                      bg.em().run(new PlayStadium(stadiumCard))
-                      bc "$stadiumCard is now a Stadium"
+                    }
+                    onRemoveFromPlay {
+                      eff.unregister()
+                      bg.em().run(new ChangeImplementation(pkmnCard, stadiumCard))
+                      moveCard(pkmnCard, thisCard.player.pbg.hand)
                     }
                   }
+                  stadiumCard.player = thisCard.player
+                  bg.em().run(new ChangeImplementation(stadiumCard, pkmnCard))
+                  if (bg.stadiumInfoStruct) {
+                    discardStadium()
+                  }
+
+                  PlayStadium.StadiumInfoStruct stadiumInfoStruct = new PlayStadium.StadiumInfoStruct()
+                  stadiumInfoStruct.stadiumCard = stadiumCard
+                  bg.setStadiumInfoStruct(stadiumInfoStruct)
+                  bg.em().activateEffect(stadiumCard)
```

Comment:
I am afraid there are some additional logic in `PlayStadium` such as remove from play registration which won't work with this setup. Why don't you run it directly?

Reply:
Does this not handle the removeFromPlay stuff?
https://github.com/flagrama/tcgone-engine-contrib/blob/062226ec23c5073af344c2f1a4800ad49fa593fa/src/tcgwars/logic/impl/gen4/Stormfront.groovy#L2946-L2954

Reply:
You are quite right, That ought to handle that. Apologies